### PR TITLE
Support more text models and tasks

### PR DESCRIPTION
**Pull request by guangy10**
### Added 7 new models based on the current trends on [HF Hub](https://huggingface.co/models) for different tasks:
 - `encoder-only` architecture: bert-family models from Transformers and Hub
 - `decoder-only` architecture: smollm from Hub
 - `encoder-decoder` architecture: T5 form Transformers

### Added 2 new tasks:
- seq2seq (e.g. translation, summarization, etc) test generation
- mask prediction

### Improved setup:
  - Updated installation to `pip install .[tests]` to install deps required by running models
  - Updated supported model list

### Improved CI:
 - Consolidated CI jobs. One test, per model, i.e. `test_modeling_<my_model>.py` which is mirroring the models and test setup on Transformers. This will help with model code patching/rewriting and testing
 - CI matrix to cover testing against released executorch from pypi and the nightly version
 - Pin to specific transformers version for better stability
 - Make CI jobs easy to read and debug
 
### Integration points in optimum-executorch:
It's a pain that we have to wait for changes being merged and included in the new release from `Transformers` in the upstream. The cycle is long. 
In this PR, I'm proposing to add the `Transformers` integration points to `optimum-executorch` as well, this will immediately unblock adding new models that doesn't require modeling code change from `Transformers` *(it's true for most of the cases)*. Of course, we will also continue landing the integration points to `Transformers` to keep improving the export coverage. 

A typical dev workflow will look like this:
  1. Add new export wrapper module to both `optimum-executorch` and `Transformers`
  2. New models in `optimum-executorch` can be enabled rapidly w/o waiting for the wrapper in `Transformers`
  3. Once the export wrapper module is available in `Transformers`, switch to use the one from `Transformers` and the same wrapper in `optimum-executorch` can be removed
  4. The export wrapper module in `Transformers` will ensure no regression on exportability for supported models, reducing the breakage when bump up `Transformers` versions in `optimum-executorch`
  
A specific example:
  1. I'm adding `Seq2SeqLMExportableModule` module to `optimum-executorch` in this PR. In parallel the same module is pending review in `transformers` https://github.com/huggingface/transformers/pull/36486
  2. We can safely merge this PR with new models onboarded
  3. Once [#36486](https://github.com/huggingface/transformers/pull/36486) is picked in new releases, we can bump the `transformers` version in `optimum-executorch` and import the one from `transformers` as the source of truth, same as the wrapper for [causal lm](https://github.com/huggingface/transformers/blob/9be4728af8bec48073ae841881d7f4e2ac3521d1/src/transformers/integrations/executorch.py#L23)

### CODE REVIEW

**Comment by guangy10** [comment]
```
/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/resource_tracker.py:254: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
```
The failure doesn't seem to be relevant and `EuroBERT` can actually work with `executorch==0.4.0` as shown on CI from a different run here: https://github.com/huggingface/optimum-executorch/actions/runs/13954658002/job/39062678395.

**Comment by guangy10** [comment]
Okay, all CIs are green now

File: optimum/executorch/modeling.py [hunk 1]
```diff
@@ -294,33 +381,33 @@ def generate(
...
             max_seq_len = self.max_cache_size
-        generated_tokens = []
 
-        # prefill
-        for i, prompt_token in enumerate(prompt_tokens):
-            logits = self.forward(
-                input_ids=torch.tensor([prompt_token], dtype=torch.long, device=self.device).unsqueeze(0),
-                cache_position=torch.tensor([i], dtype=torch.long, device=self.device),
+        # Initialize with start token (0 for T5)
+        decoder_input_ids = torch.tensor([[0]], dtype=torch.long)
```
[the remark this answers is not in the document]
Comment (guangy10) [reply optimum/executorch/modeling.py:386]: @michaelbenayoun I think I can follow up in a separate PR. There is additional fix I'm working on to make it compatible with latest version of `transformers`. Will test and squeeze this improvement in the upcoming PR.

[the remark this answers is not in the document]
Comment (michaelbenayoun) [reply optimum/executorch/modeling.py:386]: Alright sounds good to me.
